src/providers/node.rs

```diff
@@ -367,6 +437,59 @@ impl NodeProvider {
 
         all_deps
     }
+
+    pub fn is_nx_monorepo(app: &App) -> bool {
+        app.includes_file("nx.json")
+    }
+
+    pub fn get_nx_app_name(app: &App, env: &Environment) -> Result<Option<String>> {
+        if let Some(app_name) = env.get_config_variable(NX_APP_NAME_ENV_VAR) {
+            return Ok(Some(app_name));
+        }
+
+        if let Ok(nx_json) = app.read_json::<NxJson>("nx.json") {
+            if let Some(default_project) = nx_json.default_project {
+                return Ok(Some(default_project.as_str().unwrap().to_owned()));
+            }
+        }
+
+        return Err(anyhow!("Could not derive nx app to build and run. Please add a default project to your nx config or set NIXPACKS_{}", NX_APP_NAME_ENV_VAR));
+    }
+
+    pub fn get_nx_project_json_for_app(app: &App, env: &Environment) -> Result<ProjectJson> {
+        let app_name = NodeProvider::get_nx_app_name(app, env)?.unwrap();
+        let project_path = format!("./apps/{}/project.json", app_name);
+        let nx_app_project_json = app.read_json::<ProjectJson>(&project_path);
+
+        if nx_app_project_json.is_err() {
+            return Err(anyhow!(format!(
+                "Could not resolve project.json for your NX app ({}, {})",
+                app_name, project_path
+            )));
+        }
+
+        nx_app_project_json
+    }
+
+    pub fn get_nx_output_path(app: &App, env: &Environment) -> Result<String> {
+        let project_json = NodeProvider::get_nx_project_json_for_app(app, env)?;
+        if project_json.targets.build.options.output_path.is_some() {
+            Ok(project_json
+                .targets
+                .build
+                .options
+                .output_path
+                .unwrap()
```

Comment:
Please no unwrapping. Lets gracefully handle the case where the output path is not set properly.

Same with the unwrap below.

Reply:
Good shout, lmk if the changes are good. It certainly looks more readable.